### PR TITLE
Smart data structure

### PR DESCRIPTION
**Pull request by wkucjas**

### CODE REVIEW

File: src/agent/windows/WMICommunication.h [hunk 1]
```diff
@@ -18,7 +21,7 @@ class WMICommunication
 	IWbemLocator* m_initialLocatorToWMI;
 	IWbemServices* m_services;
 	IEnumWbemClassObject* m_pEnumerator;
-	std::vector<unsigned char> m_dataVector;	// In future here will be SMARTDATA structure
+	std::shared_ptr<SmartData> m_smartData;
```
Comment (Kicer86) [inline src/agent/windows/WMICommunication.h]: why shared_ptr?

File: src/common/SmartData.h [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        int status;
+        int value;
+        int worst;
+        int rawVal;
+        int rawVal2;
+    };
+
+    std::map<unsigned char, AttrData> smartData;
+
+private:
```
Comment (Kicer86) [inline src/common/SmartData.h]: not needed

File: src/agent/windows/WMICommunication.cpp [hunk 3]
```diff
@@ -207,9 +209,23 @@ bool WMICommunication::GetSMARTDataViaWMI()
 
 void WMICommunication::FeedSmartDataStructure(std::vector<BYTE>& data, const LONG& dataSize)
```
Comment (Kicer86) [inline src/agent/windows/WMICommunication.cpp]: data can be const

File: src/agent/windows/WMICommunication.cpp [hunk 4]
```diff
@@ -207,9 +209,23 @@ bool WMICommunication::GetSMARTDataViaWMI()
...
 {
-    for (int i = 0; i < dataSize; ++i)
+
+    for (int iterator = 0; data.size() > iterator; iterator += 12)
```
Comment (Kicer86) [inline src/agent/windows/WMICommunication.cpp]: iterator? what is wrong with 'i' ;)

**Review by Kicer86 — CHANGES_REQUESTED** [review]

File: src/agent/windows/WMICommunication.cpp (hunk 4 above)
Comment (wkucjas) [reply src/agent/windows/WMICommunication.cpp]: What is wrong with interator? :>

Comment (Kicer86) [reply src/agent/windows/WMICommunication.cpp]: It is **very** common to use 'i'. Maybe something like 'pos' or even 'position' is better here? But 'iterator' looks way to broad

File: src/agent/windows/WMICommunication.h [hunk 5]
```diff
@@ -18,7 +21,7 @@ class WMICommunication
 	IWbemLocator* m_initialLocatorToWMI;
 	IWbemServices* m_services;
 	IEnumWbemClassObject* m_pEnumerator;
-	std::vector<unsigned char> m_dataVector;	// In future here will be SMARTDATA structure
-	void FeedSmartDataStructure(std::vector<BYTE>& data, const LONG& dataSize);
+	std::unique_ptr<SmartData> m_smartData;
```
Comment (Kicer86) [inline src/agent/windows/WMICommunication.h]: How about SmartData m_smartData;?

**Review by Kicer86 — APPROVED** [review]

**Pull request merged**